criu/pie/restorer.c

```diff
@@ -432,6 +432,40 @@ static int restore_signals(siginfo_t *ptr, int nr, bool group)
 	return 0;
 }
 
+static int restore_cpu_affinity(struct task_restore_args *args)
+{
+	int i;
+	int pid;
+	int ret;
+	int *need_cpu_affinity;
+	cpu_set_t *cpumask;
+	cpu_set_t *allowed_cpus;
+
+	need_cpu_affinity = (int *)args->allowed_cpus;
+	if (!*need_cpu_affinity) {
```

Comment:
It seems like you are storing multiple things in 'allowed_cpus'. A comment would be nice describing what you exactly store in 'allowed_cpus'. It took me some time that it included an integer with the information if this is turned on or off and after that the actual information.

Reply:
I think it would be better to have `need_cpu_affinity` as a separate field in `struct task_restore_args`